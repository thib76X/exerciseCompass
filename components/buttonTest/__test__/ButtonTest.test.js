import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../../buttonTest/ButtonTest';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without dying', function (){
    const div = document.createElement("div");
    ReactDOM.render(<Button blabla={'dearMadam'}/>, div)
});

it('should renders good ', function () {
    const {getByTestId} = render(<Button blabla={'dearMadam'}/>)
    expect(getByTestId('test1')).toHaveTextContent("dearMadam")
});

it('should renders good ', function () {
    const {getByTestId} = render(<Button blabla={'dearMadam2'}/>)
    expect(getByTestId('test1')).toHaveTextContent("dearMadam2")
});

it('macthes with snapshot', function () {
    const smt = renderer.create(<Button blabla={'pota'}/>).toJSON();
    expect(smt).toMatchSnapshot();
});
////talk about jest, event, chai, snapshot