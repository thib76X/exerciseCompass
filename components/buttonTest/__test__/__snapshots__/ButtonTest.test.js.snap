// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macthes with snapshot 1`] = `
<div
  data-testid="test1"
>
  pota
</div>
`;
